designs/2021-break-on-parsing-errors/README.md

```diff
@@ -0,0 +1,185 @@
+- Repo: eslint/eslint
+- Start Date: 2021/1/12
+- RFC PR: https://github.com/eslint/rfcs/pull/75
+- Authors: [A-Katopodis](https://github.com/A-Katopodis)
+
+# (Break on parsing errors)
+
+## Summary
+
+The suggested change is for ESLint tp support an argument that will make the CLI exit with code 2 if any fatal errors are reported. The option will be an opt-in boolean argument called `--fatal-parse-error`.
+
+## Motivation
+
+We met with a couple of cases where we assumed a succesfull run of ESLint in CI enviroments. When ESLint wouldn't be able to read the tsconfig.json or had a wrongly configured source type the error would be a `fatal` one but the exit code would be `1`. There was no distiction between a run with no fatal errors and with fatal errors.
+
+According to the eslint docs about exit codes:
+
+- 0: Linting was successful and there are no linting errors. If the --max-warnings flag is set to n, the number of linting warnings is at most n.
+- 1: Linting was successful and there is at least one linting error, or there are more linting warnings than allowed by the --max-warnings option.
+- 2: Linting was unsuccessful due to a configuration problem or an internal error.
+
+Being able to exit with code `2` instead of `1` it will allow for some CI pipelines to better understand the results and react differently if ESLint reports any `fatal` errors. 
```

Comment:
Updated the docs.

The general outlook of it is that a CI task that simply runs eslint and outputs the results to a SARIF should succeed in case of error exit code `1` which will mean that all the files had run the rules. Instead on an exit code `2` the task would fail indicating that there are some files that had some errors that didn't allow any rules to be written.